internal/openshift/user_idler_map.go

```diff
@@ -1,50 +1,43 @@
 package openshift
 
 import (
-	"sync"
-
 	"github.com/fabric8-services/fabric8-jenkins-idler/internal/idler"
+	cmap "github.com/orcaman/concurrent-map"
 )
 
 // UserIdlerMap is a type-safe and concurrent map storing UserIdler instances keyed against the user namespace.
 type UserIdlerMap struct {
-	sync.RWMutex
-	internal map[string]*idler.UserIdler
+	internal cmap.ConcurrentMap
 }
 
 // NewUserIdlerMap creates a new instance of UnknownUsersMap.
 func NewUserIdlerMap() *UserIdlerMap {
 	return &UserIdlerMap{
-		internal: make(map[string]*idler.UserIdler),
+		internal: cmap.New(),
 	}
 }
 
 // Load returns a pointer to UnknownUsersMap keyed against the specified key.
 func (m *UserIdlerMap) Load(namespace string) (*idler.UserIdler, bool) {
-	m.RLock()
-	result, ok := m.internal[namespace]
-	m.RUnlock()
+	v, ok := m.internal.Get(namespace)
+	if !ok {
+		return nil, ok
+	}
+	result, ok := v.(*idler.UserIdler)
```

Comment:
how about just `return v.(*idler.UserIder)` ?

Reply:
What it should return is user idler and bool indicating whethere there was a value in the map for key `namespace`, however this would return a bool for whether the value is of type user idler. Shouldn't we through  an error if not ok?

Reply:
I removed that check. It was redundant as the only way to write abstracted is by us. So no need to check while retrieving the value.